src/WordPress/Hooks.php

```diff
@@ -135,13 +135,30 @@ public function purgeCacheByRelevantURLs($postId)
             }
             $wpDomain = $wpDomainList[0];
 
-            $validPostStatus = array('publish', 'trash');
+            $validPostStatus = array('publish', 'trash', 'private');
             $thisPostStatus = get_post_status($postId);
 
             if (get_permalink($postId) != true || !in_array($thisPostStatus, $validPostStatus)) {
                 return;
             }
 
+            // We don't need to purge for private posts, except when they were just transitioned from public.
+            if ('private' === $thisPostStatus) {
+                // The action that fires on transition of status is "post_updated", which will have the old version
+                // in the extra args. If that arg is not there we can bail out.
+                if (! isset($args[1])) {
+                    return;
+                }
+                /**
+                 * @var \WP_Post $before Previous version of post.
+                 */
+                list(, $before) = $args;
+
+                if ('private' === $before->post_status) {
```

Comment:
We should change to `$before->post_status !== 'private'` as we don't interested in `trash` status or any else.

Reply:
Hey @sejoker ! Sorry for losing track of this PR and replying this late, so I have to dig in my memory a bit. I think I did it to be more future proof, not sure if it's already needed in some cases. If a post transitions to the `private` status, it should definitely not be in the cache. I can image there could be cases where it somehow does not directly transition from `publish` to `private`, but it was still previously cached.